libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -184,13 +184,84 @@ function getCandidateTallies(
   candidateNameRecord: CandidateNameRecord
 ): Record<CandidateId, VxTabulation.CandidateTally> {
   const tallies: Record<CandidateId, VxTabulation.CandidateTally> = {};
+
+  // Record that maps lowercase candidate name to an array of tallies: one tally per name spelling.
+  // In the event of case-insensitive matches, we later collapse matching names into a single
+  // write-in candidate.
+  // Example:
+  // "George Washington" gets 5 write-in votes and "George washington" gets 1 write-in vote.
+  // The resulting Record is:
+  // {
+  //   "george washington": [
+  //     {id: "...", name: "George Washington", tally: 5, isWriteIn: true},
+  //     {id: "...", name: "George washington": tally: 1, isWriteIn: true}
+  //   ]
+  // }
+  const writeInCandidateNameRecord: Record<
+    string,
+    VxTabulation.CandidateTally[]
+  > = {};
+
   for (const selection of contest.ContestSelection as ResultsReporting.CandidateSelection[]) {
-    const candidateId = selection['@id'];
-    tallies[candidateId] = {
-      id: candidateId,
-      name: candidateNameRecord[candidateId],
-      tally: findTotalVoteCounts(assertDefined(selection.VoteCounts)),
-    };
+    if (selection.IsWriteIn) {
+      // ID prefix indicates to VxAdmin logic that this was a write in
+      const candidateId = `temp-write-in-${selection['@id']}`;
+
+      const name = candidateNameRecord[selection['@id']];
+      const tally: VxTabulation.CandidateTally = {
+        id: candidateId,
+        name,
+        tally: findTotalVoteCounts(assertDefined(selection.VoteCounts)),
+        isWriteIn: true,
+      };
+
+      tallies[candidateId] = tally;
+      const lowerCaseName = name.toLowerCase();
+      if (!writeInCandidateNameRecord[lowerCaseName]) {
+        writeInCandidateNameRecord[lowerCaseName] = [];
+      }
+      writeInCandidateNameRecord[lowerCaseName].push(tally);
+    } else {
+      const candidateId = selection['@id'];
+
+      const name = candidateNameRecord[selection['@id']];
+      const tally: VxTabulation.CandidateTally = {
+        id: candidateId,
+        name,
+        tally: findTotalVoteCounts(assertDefined(selection.VoteCounts)),
+      };
+
+      tallies[candidateId] = tally;
+    }
+  }
+
+  // Dedupe write-in candidates (amongst other write-ins only).
+  // Write-ins that duplicate official candidates are not deduped ie. the
+  // candidates "regular" votes will show up separately from their write-in votes.
+  for (const lowerCaseName of Object.keys(writeInCandidateNameRecord)) {
+    // Get list of tallies grouped by case-insensitive name
+    const talliesForName = writeInCandidateNameRecord[lowerCaseName];
+    // Ignore names that only have one tally object
+    if (talliesForName.length > 1) {
```

Comment:
Nit: I don't think it's necessary to ignore names with just one tally object, since the logic below will just be a no-op